core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/coff/parser/GNUPEBinaryObject64.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2023 Space Codesign Systems and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Space Codesign Systems - Initial API and implementation
+ *     QNX Software Systems - Initial CygwinPEBinaryObject class
+ *     John Dallaway - Initial GNUPEBinaryObject64 class (#361)
+ *******************************************************************************/
+package org.eclipse.cdt.utils.coff.parser;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.eclipse.cdt.core.IBinaryParser;
+import org.eclipse.cdt.utils.AR.ARHeader;
+import org.eclipse.cdt.utils.IGnuToolFactory;
+import org.eclipse.cdt.utils.Objdump;
+import org.eclipse.core.runtime.IPath;
+
+/** @since 8.2 */
+public class GNUPEBinaryObject64 extends PEBinaryObject64 {
+
+	public GNUPEBinaryObject64(IBinaryParser parser, IPath path, ARHeader header) {
+		super(parser, path, header);
+	}
+
+	public GNUPEBinaryObject64(IBinaryParser parser, IPath path, int type) {
+		super(parser, path, type);
+	}
+
+	@Override
+	public InputStream getContents() throws IOException {
```

Comment:
I would suggest to use the following content of this function instead as it, in my opinion, makes it a bit easier to see what the return value will be.

```java
		Objdump objdump = getObjdump();
		if (objdump != null) {
			try {
				byte[] contents = objdump.getOutput();
				return new ByteArrayInputStream(contents);
			} catch (IOException e) {
				// Nothing
			}
		}
		return super.getContents();
```

Reply:
@Torbjorn-Svensson this is just a copy/paste from another implementation. Is clarity or consistency more important here?

Reply:
Aha, okay. Well in that case, either would do.
I still think my suggestion is clearer, but I suppose consistency is good to have too...
How many other places is the same snippet available at?

Reply:
I copied from `GNUElfBinaryObject`. That may be the only class with that exact implementation.